rewrite-core/src/main/java/org/openrewrite/TypeReferenceProvider.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite;
+
+import org.openrewrite.trait.TypeReference;
+
+import java.util.Set;
+
+public interface TypeReferenceProvider {
```

Comment:
Should we nest this interface as `Provider` inside `TypeReference` or is it too awkward with a file name like `META-INF/services/org.openrewrite.TypeReference$Provider`?

Reply:
It does feel slightly awkward but we'll have an implementation as an example that we can point to right away, and it's also a convention so it can be explained